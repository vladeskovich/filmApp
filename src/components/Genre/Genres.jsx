import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Typography from '../Typography';
import List from '../List';

import styles from './Genres.scss';

const Genres = ({
  className,
  data,
}) => {
  const classes = classNames(styles.genre, className);

  return (
    <div className={styles.genreWrapper}>
      <List
        data={data}
        direction="horizontal"
        ordered={false}
        className={styles.genresContainer}
        itemClassName={styles.genreItem}
      >{(genre) => (
        <Typography
          className={styles.genre}
        >
          {genre.name}
        </Typography>
      )}
      </List>
    </div>

  );
};

Genres.propTypes = {
  data: PropTypes.string,
  className: PropTypes.string,
};

Genres.defaultProps = {};

export default Genres;
