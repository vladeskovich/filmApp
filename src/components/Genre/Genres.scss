.genreWrapper {
  display: inline-block;
}

.genreItem {
  margin-right: 20px;
}

.genre {
  font-size: 14px;
  font-weight: normal;
}