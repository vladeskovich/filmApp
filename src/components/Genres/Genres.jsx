import React from 'react';
import PropTypes from 'prop-types';
import Typography from '../Typography';
import List from '../List';

import styles from './Genres.scss';

const Genres = ({
  data,
}) => (
    <div className={styles.genresContainer}>
      <List
        data={data}
        direction="horizontal"
        itemClassName={styles.genreItem}
      >{(genre) => (
        <Typography
          className={styles.genre}
        >
          {genre.name}
        </Typography>
      )}
      </List>
    </div>

);

Genres.propTypes = {
  data: PropTypes.string,
  className: PropTypes.string,
};

Genres.defaultProps = {};

export default Genres;
