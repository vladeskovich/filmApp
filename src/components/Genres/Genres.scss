.genresContainer {
  display: inline-block;
  padding: 10px 0;
}

.genreItem {
  margin-right: 20px;
}

.genre {
  font-size: 14px;
  font-weight: normal;
}