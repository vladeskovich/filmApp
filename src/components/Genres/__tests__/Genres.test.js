import React from 'react';
import Genres from '../Genres';

describe('Genres', () => {
  const props = {
    data: [
      { id: 1, name: 'Action' },
    ],
  };

  it('should render Genres component with props', () => {
    const component = mount(<Genres {...props}/>);

    expect(component.find('ul').length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
