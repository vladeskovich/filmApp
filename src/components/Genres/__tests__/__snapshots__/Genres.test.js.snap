// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres should render Genres component with props 1`] = `
<Genres
  data={
    Array [
      Object {
        "id": 1,
        "name": "Action",
      },
    ]
  }
>
  <div
    className="Genres__genresContainer_32eh6Y_C"
  >
    <List
      data={
        Array [
          Object {
            "id": 1,
            "name": "Action",
          },
        ]
      }
      direction="horizontal"
      itemClassName="Genres__genreItem_1437y2lU"
      ordered={false}
    >
      <ul
        className="List__list_3qljzsty List__horizontal_1mt9lNJ-"
      >
        <li
          className="Genres__genreItem_1437y2lU"
          key="1"
        >
          <Typography
            className="Genres__genre_25L8dYmX"
            variant="body"
          >
            <p
              className="Typography__typography_1E76fqYp Genres__genre_25L8dYmX"
            >
              Action
            </p>
          </Typography>
        </li>
      </ul>
    </List>
  </div>
</Genres>
`;
